### ISSUE TITLE
Token expiration with custom API

### ISSUE THREAD

**Issue by vadimshvetsov** [danschultzer/pow#552]
Hi! Thanks for a great library.

In Phoenix umbrella app I have custom API token authentication setup based on [this guide](https://github.com/danschultzer/pow/blob/master/guides/api.md) (with renew) for using with Absinthe (socket + http) and React/Apollo on the frontend. I can't understand when api and renew token expires.

I've tried to find `max_age` settings for `api_token` (and `api_renew_token`) and actually can't find any defaults and how to change them with this config, looks like there is no difference between api and renew `max_age`. 

```ex
config :proling_web, :pow,
  repo: Proling.Repo,
  user: Proling.Identity.User,
  mailer_backend: ProlingWeb.Mailer,
  web_mailer_module: ProlingWeb,
  message_verifier: {Pow.Plug.MessageVerifier, key_generator_opts: [max_age: 10]},
  cache_store_backend: Pow.Store.Backend.MnesiaCache,
  extensions: [PowResetPassword, PowEmailConfirmation],
  controller_callbacks: Pow.Extension.Phoenix.ControllerCallbacks
```

For signing token I call `Pow.Plug.sign_token(conn, signing_salt(), token, config)`.  It uses explicit `Pow.Plug.MessageVerifier` which ignores `max_age` option further in `KeyGenerator`. In high level `Plug.Crypto` API we can encode `max_age` to a token.

https://github.com/elixir-plug/plug_crypto/blob/3fbecf0d9e5ba10214f0ea527c2cc13c929ef8d0/lib/plug/crypto.ex#L168

Is there any solution for setting `max_age` without adding rewritten custom module instead of `Pow.Plug.MessageVerifier`?

I'm open for PR with adding `max_age` to `Pow.Plug.MessageVerifier` but maybe there is a reason to not adding `max_age` or not using `Plug.Crypto` high level API with first implementation and I'm wrong?

**Comment by danschultzer** [comment]
In Pow the tokens are not stateless so there's no need for an encoded `max_age` value as they automatically will be expired by the cache `ttl` setting. For renewal token it's the default for the `PowPersistentSession.Store.PersistentSessionCache` [which is 30 days](https://github.com/danschultzer/pow/blob/v1.0.20/lib/extensions/persistent_session/store/persistent_session_cache.ex#L4).

The plug signing mechanism is usually used for stateless tokens where a max age would be necessary.

All you have to do is add `:ttl` to the store config:

```elixir
credentials_store_config = Keyword.put(store_config, :ttl, :timer.minutes(15))
persistent_session_store_config = Keyword.put(store_config, :ttl, :timer.hours(24))

CredentialsCache.put(credentials_store_config, access_token, {user, [renewal_token: renewal_token]})
PersistentSessionCache.put(persistent_session_store_config, renewal_token, {[id: user.id], [access_token: access_token]})
```

I think it would be great to improve the docs regarding so, do you see anywhere where this information would fit and can help others in the same situation? I could maybe just add the above to the API guide as comments to highlight how you adjust the store config.

**Comment by vadimshvetsov** [comment]
@danschultzer Dan, thanks a lot, this is super helpful and complete explanation. I've tried `persistent_session_ttl` parameter but it works only with `Pow.Plug.Session` is I've found out. I think your answer could be mentioned as a paragraph in API guide before [`OAuth 2.0`](https://github.com/danschultzer/pow/blob/master/guides/api.md) because using this parameters is essential for testing token renewal mechanics in custom GraphQL API for example.
